File: problems/346/solution.cpp
class MovingAverage {
private:
    int size;
    queue<int> q;
    double cur;
public:
    /** Initialize your data structure here. */
    MovingAverage(int _size): size{_size}, cur{0.0} {}
    
    double next(int val) {        
        if (q.size() == size) {
            cur -= q.front();
            q.pop();
        }
        
        cur += val;
        
        q.push(val);
        
        return cur / q.size();
    }
};

/**
 * Your MovingAverage object will be instantiated and called as such:
 * MovingAverage* obj = new MovingAverage(size);
 * double param_1 = obj->next(val);
 */

File: problems/346/solution.py
class MovingAverage:

    def __init__(self, size: int):
        """
        Initialize your data structure here.
        """
        self.size = size
        self.buf = []
        

    def next(self, val: int) -> float:
        if len(self.buf) >= self.size:
            self.buf.pop(0)
        
        self.buf.append(val)
        
        return sum(self.buf) / len(self.buf)


# Your MovingAverage object will be instantiated and called as such:
# obj = MovingAverage(size)
# param_1 = obj.next(val)